src/OpenTelemetry.Collector.Dependencies/Implementation/AzureSdkDiagnosticListener.cs

```diff
@@ -115,5 +120,26 @@ public override void OnException(Activity current, object valueValue)
 
             span.Status = Status.Unknown.WithDescription(valueValue?.ToString());
         }
+
+        private string GetOperationName(Activity activity)
+        {
+            // activity name looks like 'Azure.<...>.<Class>.<Name>'
+            // as namespace is too verbose, we'll just take the last two nodes from the activity name as telemetry name
+            string activityName = activity.OperationName;
+            int methodDotIndex = activityName.LastIndexOf('.');
+            if (methodDotIndex <= 0)
```

Comment:
why this check is `<= 0` and one below `== -1`? Do you expect the name starting with dot?

Reply:
This is a way to prevent next line to throw if `methodDotIndex == 0`

`int classDotIndex = activityName.LastIndexOf('.', methodDotIndex - 1);`

I.e. activity name that starts with `.` is not what we expect, but if happened it's easier to allow it rather than complicate code further.